pandas/tests/indexes/multi/test_setops.py

```diff
@@ -555,7 +555,8 @@ def test_union_nan_got_duplicated():
     mi1 = MultiIndex.from_arrays([[1.0, np.nan], [2, 3]])
     mi2 = MultiIndex.from_arrays([[1.0, np.nan, 3.0], [2, 3, 4]])
     result = mi1.union(mi2)
-    tm.assert_index_equal(result, mi2)
+    expected = MultiIndex.from_arrays([[1.0, 3.0, np.nan], [2, 4, 3]])
```

Comment:
@phofl was this addressed in 1.5 or on main? GIven the base `sort=None` documentation I suppose this is correct.

Could you add a test for `sort=False` too?

Reply:
The current (no-sort) behavior has been around since at least 1.3.0 when this test was added. I parameterized the test with sort=None|False and added a whatsnew.

Reply:
Yeah the new behavior is correct, the goal back then was something different than the sorting behavior